plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeAnalyze.java

```diff
@@ -479,6 +469,70 @@ public void testStatsOnTpcDsData()
         }
     }
 
+    @Test
+    public void testCreateTableStatisticsWhenCollectionOnWriteDisabled()
+    {
+        String tableName = "test_statistics_" + randomNameSuffix();
+        assertUpdate(
+                disableStatisticsCollectionOnWrite(getSession()),
+                "CREATE TABLE " + tableName + " AS SELECT * FROM tpch.sf1.nation",
+                25);
+
+        assertQuery(
+                "SHOW STATS FOR " + tableName,
+                "VALUES " +
+                        "('nationkey', null, null, 0.0, null, 0, 24)," +
+                        "('regionkey', null, null, 0.0, null, 0, 4)," +
+                        "('comment', null, null, 0.0, null, null, null)," +
+                        "('name', null, null, 0.0, null, null, null)," +
+                        "(null, null, null, null, 25.0, null, null)");
+
+        runAnalyzeVerifySplitCount(tableName, 1);
+
+        assertQuery(
+                "SHOW STATS FOR " + tableName,
+                "VALUES " +
+                        "('nationkey', null, 25.0, 0.0, null, 0, 24)," +
+                        "('regionkey', null, 5.0, 0.0, null, 0, 4)," +
+                        "('comment', 1857.0, 25.0, 0.0, null, null, null)," +
+                        "('name', 177.0, 25.0, 0.0, null, null, null)," +
+                        "(null, null, null, null, 25.0, null, null)");
+    }
+
+    @Test
+    public void testCreatePartitionedTableStatisticsWhenCollectionOnWriteDisabled()
+    {
+        String tableName = "test_statistics_" + randomNameSuffix();
+        assertUpdate(
+                disableStatisticsCollectionOnWrite(getSession()),
+                "CREATE TABLE " + tableName
+                        + " WITH ("
+                        + "   partitioned_by = ARRAY['regionkey']"
+                        + ")"
+                        + "AS SELECT * FROM tpch.sf1.nation",
+                25);
+
+        assertQuery(
+                "SHOW STATS FOR " + tableName,
+                "VALUES " +
+                        "('nationkey', null, null, 0.0, null, 0, 24)," +
+                        "('regionkey', null, 5.0, 0.0, null, null, null)," +
+                        "('comment', null, null, 0.0, null, null, null)," +
+                        "('name', null, null, 0.0, null, null, null)," +
+                        "(null, null, null, null, 25.0, null, null)");
+
+        runAnalyzeVerifySplitCount(tableName, 5);
```

Comment:
i know it's preexisting but i don't think we need to assert split count in every test method here. It blurs the test's intent
(perhaps, we don't need it in any test, i don't know, but i am not requesting any change to existing tests)

this would be better:

```
assertUpdate("ANALYZE " + tableName);
```